Rubberduck.Inspections/Concrete/ParameterCanBeByValInspection.cs

```diff
@@ -125,29 +124,16 @@ private static bool IsUsedAsByRefParam(IEnumerable<Declaration> declarations, De
                 .GroupBy(call => call.Declaration)
                 .ToList(); // only check a procedure once. its declaration doesn't change if it's called 20 times anyway.
 
-            foreach (var item in procedureCalls)
-            {
-                var calledProcedureArgs = items
-                    .Where(arg => arg.DeclarationType == DeclarationType.Parameter && arg.ParentScope == item.Key.Scope)
-                    .OrderBy(arg => arg.Selection.StartLine)
-                    .ThenBy(arg => arg.Selection.StartColumn)
-                    .ToArray();
-
-                foreach (var declaration in calledProcedureArgs)
-                {
-                    if (((VBAParser.ArgContext)declaration.Context).BYVAL() != null)
-                    {
-                        continue;
-                    }
-
-                    if (declaration.References.Any(reference => reference.IsAssignment))
-                    {
-                        return true;
-                    }
-                }
-            }
-
-            return false;
+            //TODO convert to method syntax
+            return (from item in procedureCalls
+                        from declaration in items
+                            .Where(arg => arg.DeclarationType == DeclarationType.Parameter && arg.ParentScope == item.Key.Scope)
+                            .OrderBy(arg => arg.Selection.StartLine)
+                            .ThenBy(arg => arg.Selection.StartColumn)
+                            .ToArray()
+                        where ((VBAParser.ArgContext)declaration.Context).BYVAL() == null
+                        select declaration)
+                    .Any(dec => dec.References.Any(reference => reference.IsAssignment));
```

Comment:
I fail to see how this is more clear than the original. And it is certainly slower.